### PR TITLE
db/DataPoolConnectorOptions

### PR DESCRIPTION
**Pull request by karolkolodziej**
Added documentation for `DataPoolConnectorOptions`.

### CODE REVIEW

File: ts/Data/DataPoolOptions.ts [hunk 1]
```diff
@@ -33,8 +33,22 @@ export interface DataPoolOptions {
 
 export interface DataPoolConnectorOptions
 <T extends keyof DataConnectorTypes = keyof DataConnectorTypes> {
+    /**
+     * The unique identifier of the connector. Used later when referencing
+     * the connector in the component where it is used.
+     **/
     id: string;
+    /**
+     * The options of the given connector type.
```
Comment (sebastianbochan) [inline ts/Data/DataPoolOptions.ts:42]: Would be good to add an example of using it.

File: ts/Data/DataPoolOptions.ts [hunk 2]
```diff
@@ -33,8 +33,22 @@ export interface DataPoolOptions {
...
+     **/
     options: DataConnectorTypes[T]['prototype']['options'];
+    /**
+     * The type of the connector, depends on your data source.
+     * Possible values are:
+     * - `CSV`
+     * - `GoogleSheets`
+     * - `HTMLTable`
```
Comment (pawellysy) [inline ts/Data/DataPoolOptions.ts:50]: Isn't the bundler show all available types?

**Review by pawellysy — APPROVED** [review]
LGTM! One minor comment tho

File: ts/Data/DataPoolOptions.ts (hunk 2 above)
Comment (karolkolodziej) [reply ts/Data/DataPoolOptions.ts:50]: No, there is no registry for that.

Comment (bre1470) [reply ts/Data/DataPoolOptions.ts:62]: `DataConnector.types` is it.

File: ts/Data/DataPoolOptions.ts [hunk 3]
```diff
@@ -33,8 +33,34 @@ export interface DataPoolOptions {
...
+     *       }]
+     * },
+     **/
     options: DataConnectorTypes[T]['prototype']['options'];
+    /**
+     * The type of the connector, depends on your data source.
+     * Possible values are:
+     * - `CSV`
+     * - `GoogleSheets`
+     * - `HTMLTable`
```
Comment (bre1470) [inline ts/Data/DataPoolOptions.ts:62]: Please replace with something like
> The available types can be found in the `DataConnector.types` registry.

**Review by bre1470 — CHANGES_REQUESTED** [review]

**Review by sebastianbochan — APPROVED** [review]

**Review by pawelfus — CHANGES_REQUESTED** [review]
Asana: "... and Connector options" 😬 

Zrzut ekranu 2023-06-28 o 17 00 50

Question: Can we define it? So we have access to `GoogleSheetsConnector.Options`, `HTMLTableConnector.Options` and `CSVConnector.Options`?

**Comment by bre1470** [comment]
We can link them in the doclet of the class property.

**Review by pawelfus — APPROVED** [review]
Looks good (except the part with `DataPoolConnectorOptions.options`)

File: ts/Data/DataPoolOptions.ts (hunk 2 above)
Comment (pawelfus) [reply ts/Data/DataPoolOptions.ts:72]: +1

File: ts/Data/DataPoolOptions.ts [hunk 4]
```diff
@@ -28,13 +31,46 @@ import type { DataConnectorTypes } from './Connectors/DataConnectorType.js';
  * */
 
 export interface DataPoolOptions {
+    /**
+     * Link list to connector options + mention DataConnector.types
+     *
```
Comment (bre1470) [inline ts/Data/DataPoolOptions.ts:36]: This was text for your eyes only and should be replaced. :)

File: ts/Data/DataPoolOptions.ts [hunk 5]
```diff
@@ -28,13 +31,46 @@ import type { DataConnectorTypes } from './Connectors/DataConnectorType.js';
...
+     * Available connectors and its options:
+     * {@link CSVConnectorOptions | CSVConnector}
+     * {@link GoogleSheetsConnectorOptions | GoogleSheetsConnector}
+     * {@link HTMLTableConnectorOptions | HTMLTableConnector}
```
Comment (bre1470) [inline ts/Data/DataPoolOptions.ts:40]: This should be formatted as a list.

File: ts/Data/Connectors/DataConnector.ts [hunk 6]
```diff
@@ -367,8 +367,17 @@ namespace DataConnector {
...
+         * Option of the modifier that is used to modify the data.
+         */
         dataModifier?: DataModifierTypeOptions;
+        /**
+         * Options for the data table.
+         */
         dataTable?: DataTable.Options;
+        /**
+         * Your custom metadata.
+         */
```
Comment (bre1470) [inline ts/Data/Connectors/DataConnector.ts:380]: This one should also become a separate file.

**Review by bre1470 — CHANGES_REQUESTED** [review]

File: ts/Data/DataPoolOptions.ts (hunk 4 above)
Comment (karolkolodziej) [reply ts/Data/DataPoolOptions.ts:36]: 🙈

File: ts/Data/DataPoolOptions.ts (hunk 2 above)
Comment (bre1470) [reply ts/Data/DataPoolOptions.ts:74]: DataConnector registry is not ready for API docs. So we leave it for now.

File: ts/Data/Connectors/DataConnectorOptions.d.ts [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+    /**
+     * Option of the modifier that is used to modify the data.
+     */
+    dataModifier?: DataModifierTypeOptions;
+    /**
+     * Options for the data table.
+     */
+    dataTable?: DataTable.Options;
+    /**
+     * Your custom metadata.
```
Comment (bre1470) [inline ts/Data/Connectors/DataConnectorOptions.d.ts:42]: This needs more explanation. Something like
> Contains custom metadata related to the connector source, for example the column order on the source side.

File: ts/Data/Connectors/DataConnectorOptions.d.ts [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+     */
+    metadata?: Metadata;
+}
+
+/**
+ * Metadata
```
Comment (bre1470) [inline ts/Data/Connectors/DataConnectorOptions.d.ts:48]: Same here.

File: ts/Data/Connectors/DataConnectorOptions.d.ts [hunk 9]
```diff
@@ -0,0 +1,70 @@
...
+ */
+export interface Metadata {
+    columns: Record<string, MetaColumn>;
+}
+
+/**
+ * Metadata entry containing the name of the column and a metadata object.
```
Comment (bre1470) [inline ts/Data/Connectors/DataConnectorOptions.d.ts:55]: This is misplaced and should go to `Metadata.columns`. Here maybe something like:
> Metadata for a specific column in the connector source.

**Review by bre1470 — CHANGES_REQUESTED** [review]

**Review by bre1470 — APPROVED** [review]
Thank you! Looks good. 👍

**Review by goransle — APPROVED** [review]

**Pull request merged**